ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml

```diff
@@ -74,13 +60,40 @@
     - xlc_v13
     - xlc_v16
 
+    # 4. Core OSS installation
+
+    ## Note: this may actually already be performed - partially
+    ## depending on how python - to use Ansible - has been prepared
+    # yum based installation of additional software
+    # installs some gnu packages required by ansible modules
+    # Note: AIX File system configuration must be run now as RPM based software
+    # cannot expand filesystem space on demand
+    - aixfs
+    - yum
+
     ## additional OSS packages
     - ant
     - ant_contrib
 
+    # 5. AIX User Admin
+
+    # users needed - wait until here to be sure bash is installed
+    # tbd: Need other actions performed first - mainly adding /usr/bin/bash
+    # to the list of legal shells.
+    - role: rbac
+      tags:
+        - adoptopenjdk
+        - users
+    - role: users
```

Comment:
- Actually, what I am trying to achieve here is being able to specify a username (e.g., `aixtools` as username) on the command-line using `ansible-playbook --tags=users -e "users=aixtools" ..../main.yml` because my local user for building needs the settings `jenkins` needs.
- what I need to add to the documentation (when I learn where it is to go, regular FAQ.md, or a special one) - add the need for: `touch roles/users/tasks/aixtools.yml`
- I'll add more lower-level tags such as:
```
./tasks/jenkins.yml:
- name: warn about failed copy
  debug:
    msg: "{{ Jenkins_User_SSHKey }} failed to copy"
  when: jenkins_rsa.failed == true
  tags: jenkins_user
- Last, considering the instructions is FAQ.md:
`ansible-playbook -b -i inventory_file --skip-tags adoptopenjdk,jenkins_user playbooks/AdoptOpenJDK_Unix_Playbook/main.yml`
The tag `adoptopenjdk` before the on the users role - skips all the users creations - including jenkins.

Reply:
> Hmmm I'm going to need to be convinced on this one. Three reasons:
> 
>     * This makes the process inconsistent between 
a) Maybe the only things *nix's have in common is _nix_ in the branding.
b) the bigger issue is that we are still in the process of transforming the flatfile `aix.yml` into proper ansible structure(s).
c) copy/paste a bunch of lines from `aix.yml' to roles/somerolename/tasks/main.yml does not make it a _proper_ role.
> AIX and other UNIX platforms which makes maintenance harder so I wouldn't want to accept this unless it was proposed across all OS playbooks (same for the use of the `adoptium` group)
> 
- removed the adoptium group. As I mentioned above - this is a `security` consideration to not have everything in a default group. However, as adoptium internal systems are not really commercial production systems - I'll skip it. If/when this were to be run on a non-Ansible project system I would consider the group more important.
- as to unfying the playbooks - I have glanced at the *Unix* playbooks and see many opportunities for making better use of Ansible constructs AND I see the *AIX* playbooks as a relatively safe way to gain experience with using code such as:
```
- name: Include Platform specific variables
  include_vars: "{{ansible_os_family}}".yml"
- name: include Platform specific configuration tasks
  include_tasks: "config-{{ ansible-os-family }}.yml"
```
And in a proper role these would be located in `roles/rolename/{vars|tasks}/` respectively.
- imho: having all the users in one area is cleaner, and easier to maintain. And a proper role (user management, versus task after task after task to do, or not do users.
-  I cannot speak re: the *Unix* users, because I have not looked at them, but with the AIX users one change - especially with ulimits - was three edits.
- re: ease of maintaining users: it took a few passes to get all the common material centralized, ie.e., what was common, what was unique was not something jumping out at you.
- having all user information in one area (directory) makes, imho, triage of user issues much easier.
- ansible has it's idiosyncrasies - and where I had the most difficulty was finding a good way to get _loops_ and _tags_ to work.
>     * I think there's enough diffferences between the users that this is best kept as separate roles. The requirements for each user are fairly different and the `zeus` user in particular should be as basic/clean as possible.
- In AIX - ulimits are ulimits, don't care what your username is.
- The 'customization' for zeus, if you want that is to remove the non-ulimits attributes.
- The nagios 'user' confused me, but I left it there - **note** on the adopt nodes only three of the 8 have a authorized_key that exists AND is not zero bytes.
- In any case, I feel it is much more `ansible way` to break things into individual playbooks to perform one task
- one book over ansible gives (design) advice to keep individual files to under 100 lines - as being more modular
- ansible `roles` are not merely a way to move tasks out of ./main.yml to a sub-directory, but to organize defaults and vars in the role.
- (I spent a lot of time researching Ansible to come up with this model).
> 
>     * Most of the special options that we add to the users for various permissions are only required on the `jenkins` user so unless there's a reason to I wouldn't want to add them to the others (I think these roles are applying a lot of the settings to all of them)
# Anyway
- awaiting your reply.